File: numeric/euclidgcd/euclidgcd.r

euclidgcd <- function(m, n) {
  r <- m %% n
  while (r != 0) {
    m <- n
    n <- r
    r <- m %% n
  }
  n
}

v_1 <- 15
v_2 <- 10

args <- commandArgs(trailingOnly = TRUE)
if (length(args) >= 2) {
  v_1 <- as.integer(args[1])
  v_2 <- as.integer(args[2])
}

sprintf("%d %d", v_1, v_2)
sprintf("%d", euclidgcd(v_1, v_2))

File: numeric/euclidgcd/euclidgcd.f90
program EuclidGcdApp
    implicit none

    integer :: m, n, gcd, num_args
    character(len=100) :: buffer

    m = 15
    n = 10

    num_args = command_argument_count()
    if (num_args > 1) then
        call get_command_argument(1, buffer)
        read(buffer, *) m
        call get_command_argument(2, buffer)
        read(buffer, *) n
    end if

    gcd = euclidgcd(m, n)

    print '(I0, " ", I0)', m, n
    print '("gcd: ", I0)', gcd

contains

    function euclidgcd(m_in, n_in) result(gcd)
        integer, intent(in) :: m_in, n_in
        integer :: m, n, gcd

        m = m_in
        n = n_in
        gcd = mod(m, n)
        do while (gcd /= 0)
            m = n
            n = gcd
            gcd = mod(m, n)
        end do
        gcd = n
    end function
end program EuclidGcdApp
